pyproject.toml

```diff
@@ -23,7 +23,7 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Mathematics",
 ]
 urls = {repository = "https://github.com/patrick-kidger/diffrax" }
-dependencies = ["jax>=0.4.23", "jaxtyping>=0.2.24", "typing_extensions>=4.5.0", "equinox>=0.11.2", "lineax>=0.0.4", "optimistix>=0.0.6"]
+dependencies = ["jax>=0.4.23", "jaxtyping>=0.2.24", "typing_extensions>=4.5.0", "typeguard==2.13.3", "equinox>=0.11.2", "lineax>=0.0.4", "optimistix>=0.0.6"]
```

Comment:
Why is `typeguard` pinned to an exact version, especially one that is so old?

It's generally considered very bad packaging practice to pin to exact versions as it's ~never required and prevents getting even security patches without publishing a new version.

Also, it's ancient and won't be getting any updates - current HEAD is 4.2.1:


![image](https://github.com/patrick-kidger/diffrax/assets/881019/aedd5652-d344-4864-8644-1555088559b9)

![image](https://github.com/patrick-kidger/diffrax/assets/881019/1c751fd2-6132-413a-b575-c4900329d5b8)